src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class MvidReader : BinaryReader
```

Comment:
Are there callers that instantiate this type? If not, could this be a `static` class (with a single `public` method) that operates on a `BinaryReader` directly?